docs/DeveloperGuide.md

```diff
@@ -500,6 +501,72 @@ For phone, even if there is a difference in white space, it is still considered
+fill-in the textbox with "find n/Alice", and then pressing down-arrow key will
+fill-in textbox with "find n/Bob" again.
+
+**Picture explain:**<br>
+**Step 0, no user input yet:**
+![CommandListState0](images/CommandList0.png)
+When the CommandList is empty, the pointer will point to position 0.
+
+**Step 1, User executed "find n/Alice":**
+![CommandListState1](images/CommandList1.png)
+When the CommandList has one history, the pointer will point to position 1,
+which is the next position and is empty.
+
+**Step 2, User executed "find n/Bob":**
+![CommandListState2](images/CommandList2.png)
+When the CommandList has two history, the pointer will point to position 2,
+which is the next position and is empty.
+
+**Step 3, User pressed up-arrow key:**
+"find n/Bob" is fetched and auto-filled in textbox
+![CommandListState4](images/CommandList3.png)
+When the user presses up-arrow key, CommandList will decrease the pointer by one.
+Now the pointer points to position 1, which is the previous input. Then the information
+pointed by pointer will be fetched and auto-filled into the command-box.
+
+**Step 5, User pressed up-arrow key again:**
+"find n/Alice" is fetched and auto-filled in textbox
+![CommandListState5](images/CommandList4.png)
+When the user presses up-arrow key again, CommandList will decrease the pointer by one.
+Now the pointer points to position 1, which is the previous input. Then the information
+pointed by pointer will be fetched and auto-filled into the command-box.
+
+**Step 6, User pressed down-arrow key:**
+"find n/Bob" is fetched and auto-filled in textbox
+![CommandListState6](images/CommandList5.png)
+When the user presses down-arrow key, CommandList will increase the pointer by one.
+Now the pointer points to position 2, which is the next input. Then the information
+pointed by pointer will be fetched and auto-filled into the command-box.
+
+To conclude, when user have not yet press the up-arrow key, the pointer will always point to
+the next position of CommandList. So that when user presses the up-arrow key, pointer will decrease
+by one to point to the previous position and fetch the history.
+When There is no previous/next command available, a CommandException will be thrown
+and error message will be shown in the message-box.
+<br><br>
+**Aspect: The execution of PreviousCommand/NextCommand:**
+
+When a `PreviousCommand` or `NextCommand` is being executed, if it is executed successfully,
+it will return a special `CommandResult`(with `CommandRemark` set to `HISTORY`) to inform `UI` and ask `UI` to auto-fill the textbox
+with the most recent Command. If it is not executed successfully(i.e. there is no previous
+or next command available), it will throw `CommandException` and show the error message.
+
+The following activity diagram summarizes what happens when a user executes PreviousCommand/NextCommand:
+
+<img src="images/HistoryActivityDiagram.png" width="482" />
```

Comment:
Shouldn't the diamond in activity diagrams not contain words?

In addition:
<img width="679" alt="Screen Shot 2022-04-10 at 9 10 47 PM" src="https://user-images.githubusercontent.com/22218752/162619850-5d1cff6e-c2ce-46b4-a0d1-6ed3616c5357.png">